arrows/ocv/detect_motion_mog2.h

```diff
@@ -0,0 +1,110 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief OCV motion detection mog2 algorithm impl interface
+ */
+
+#ifndef KWIVER_ARROWS_OCV_DETECT_MOTION_MOG2_H_
+#define KWIVER_ARROWS_OCV_DETECT_MOTION_MOG2_H_
+
+#include <memory>
+
+#include <opencv2/opencv.hpp>
+
+#include <vital/types/timestamp.h>
+#include <vital/vital_config.h>
+#include <vital/algo/detect_motion.h>
+#include <vital/config/config_block.h>
+
+#include <arrows/ocv/kwiver_algo_ocv_export.h>
+
+namespace kwiver {
+namespace arrows {
+namespace ocv {
+
+/// OCV implementation of detect_motion_using cv::BackgroundSubtractormog2
+class KWIVER_ALGO_OCV_EXPORT detect_motion_mog2
+  : public vital::algorithm_impl<detect_motion_mog2,
+                                 vital::algo::detect_motion>
+{
+public:
+  PLUGIN_INFO( "detect_motion_mog2",
```

Comment:
This name does not have to be so long. It could be shortened to "ocv_mog2"

Reply:
+1

Reply:
Changed to ocv_mog2